addons/barcodes/models/barcode_nomenclature.py

```diff
@@ -17,18 +11,13 @@
     ('always','Always'),
 ]
 
-class barcode_nomenclature(osv.osv):
+class BarcodeNomenclature(models.Model):
     _name = 'barcode.nomenclature'
-    _columns = {
-        'name': fields.char('Nomenclature Name', size=32, required=True, help='An internal identification of the barcode nomenclature'),
-        'rule_ids': fields.one2many('barcode.rule','barcode_nomenclature_id','Rules', help='The list of barcode rules'),
-        'upc_ean_conv': fields.selection(UPC_EAN_CONVERSIONS, 'UPC/EAN Conversion', required=True,
-            help='UPC Codes can be converted to EAN by prefixing them with a zero. This setting determines if a UPC/EAN barcode should be automatically converted in one way or another when trying to match a rule with the other encoding.'),
-    }
 
-    _defaults = {
-        'upc_ean_conv': 'always',
-    }
+    name = fields.Char(string='Nomenclature Name', required=True, help='An internal identification of the barcode nomenclature')
+    rule_ids = fields.One2many('barcode.rule', 'barcode_nomenclature_id', string='Rules', help='The list of barcode rules')
+    upc_ean_conv = fields.Selection(UPC_EAN_CONVERSIONS, string='UPC/EAN Conversion', required=True, default='always',
+        help="UPC Codes can be converted to EAN by prefixing them with a zero. This setting determines if a UPC/EAN barcode should be automatically converted in one way or another when trying to match a rule with the other encoding.")
 
     # returns the checksum of the ean13, or -1 if the ean has not the correct length, ean must be a string
     def ean_checksum(self, ean):
```

Comment:
Hey man migration means not to remove
```
cr
uid
ids
context
```
Odoo have some standards, please stick with it.
In migration need some technical improvements you have to do it.
In below method
```
def ean_checksum(self, ean):
def ean8_checksum(self,ean):
```
First variable ```code```  is use less so please remove it and directly do it like : 
 ```
if len(list(ean)) != 13:
    return -1
```
![beaprogrammer](https://cloud.githubusercontent.com/assets/14940179/12079725/1f2f8294-b201-11e5-819f-34955f9cb6ca.png)